### ISSUE TITLE
The quality of grasp poses annotatation for each object using a pretrained model RAGT-3/3 is not very good, how to improve it?

### ISSUE THREAD

**Issue by andyoung009** [kmittle/Grasp-Detection-NBMOD#2]
Hi, it is a nice work. I am interested in it.  I used the RAGT-3/3 pretrain model to test on a random scene picture, the results were not so good! Is there domain gap between the sence in your dataset generated by the scene description and the RAGT-3/3 dataset?Would you please give some details on this pipeline? Thanks!
<img width="147" alt="1" src="https://github.com/kmittle/Grasp-Detection-NBMOD/assets/125349112/a7dc9aa2-51af-433e-bd5b-d459ac2355c9">

**Comment by kmittle** [comment]
The models in this article were trained on NBMOD, and the targets for the two datasets are different. You need to retrain the models using your dataset.

**Comment by kmittle** [comment]
The models in the paper were trained on NBMOD, and the targets for the two datasets are different. You need to retrain the models using your dataset.

**Comment by andyoung009** [comment]
Thanks for your reply！
Did you mean that the gap between the data I use for inference here and the dataset used for training your pre-training model is relatively large? So I need to use my dataset to fine-tune the model and then perform inference verification?
The author of the paper 'Grasp-Anything: Large-scale Grasp Dataset from Foundation Models' said they use the pretrained model of RAGT-3/3 to annotate grasp poses for each object in the grasp list of the scene description generated by ChatGPT and images generated by Stable Diffusion 2.1 that align with the scene descriptions.
If there is gap between the dataset, how could they use the pretrained model?
Hope your reply, thanks!

**Comment by kmittle** [comment]
I think your question is excellent!
In fact, I have also read the paper you mentioned and noticed this point you raised. I assume the authors may have fine-tuned the model using the Jacquard dataset or the Cornell dataset. Otherwise, directly inferring pseudo-labels using the pre-trained weights of RAGT-3/3 could result in low label quality.
For specific details, you might need to consult the authors of "Grasp Anything."

**Comment by andyoung009** [comment]
Thanks for your reply！
I also connected the author of "Grasp Anything" in this [issue](https://github.com/andvg3/Grasp-Anything/issues/7), they give the answer that "there is a hypothesis in our experiment section: "Since our dataset is generated by the foundation models, will it work on real robot experiments?" We demonstrate that policy that is trained on Grasp-Anything can result in meaningful grasp operations in real robot experiments." Before generating grasp poses of a specific object in the scene, we mask out the object first and then use the masked image to generate grasp poses.
I think there may be something I didn't get.
Once again, thanks to all of your replies.

**Comment by kmittle** [comment]
You're welcome. Perhaps you could try annotating pseudo-labels using the weights of RAGT-3/3 once "Grasp Anything" is open-sourced, but I think the results might not be good if fine-tuning is not done on datasets like Cornell.
